components/Buttons/src/MDCButton.m

```diff
@@ -676,6 +681,31 @@ - (void)updateTitleColor {
   }
 }
 
+- (BOOL)mdc_adjustsFontForContentSizeCategory {
+  return _mdc_adjustsFontForContentSizeCategory;
+}
+
+- (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
+  _mdc_adjustsFontForContentSizeCategory = adjusts;
+  if (_mdc_adjustsFontForContentSizeCategory) {
+    UIFont *font = [UIFont mdc_preferredFontForMaterialTextStyle:MDCFontTextStyleButton];
+    self.titleLabel.font = font;
+    [[NSNotificationCenter defaultCenter] addObserver:self
```

Comment:
We are not concerned about the order of operations.
Explicitly setting the font handles the case where the dev set adjustFont to YES but the user doesn't actually adjust their setting.